modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/js/ckeditor5/plugins/HeadlessItemSelector.ts

```diff
@@ -0,0 +1,197 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {IFrontendDataSetProps} from '@liferay/frontend-data-set-web';
+import {ButtonView, Command, Plugin} from 'ckeditor5';
+import {openItemSelectorModal} from 'frontend-js-item-selector-web';
+
+import getIcon from '../utils/getIcon';
+
+const FDS_DEFAULT_PROPS: Partial<IFrontendDataSetProps> = {
+	pagination: {
+		deltas: [{label: 20}, {label: 40}, {label: 60}],
+		initialDelta: 20,
+	},
+	selectionType: 'single',
+};
+
+const CMS_FILE_VIEWS = [
```

Comment:
All of this item selector configuration is a copy of [configuration](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-item-selector-sample-web/src/main/resources/META-INF/resources/js/utils/defaultViews.ts) from `frontend-js-item-selector-sample-web`. Since that is a sample module, it didn't feel right to me to export it from there. I don't think there is a need for DRY here, configurations can change over time.